### PR TITLE
Fix book validates && remove quantity and is pdf

### PR DESCRIPTION
**Pull request by fadyyousry**

### CODE REVIEW

File: app/models/book.rb [hunk 1]
```diff
@@ -39,7 +48,7 @@ def create_authors authors_params
     cascade_authors
     self.authors.clear
     authors_params[:names].split(',').each do |value|
-      author = Author.find_or_initialize_by({name: value})
+      author = Author.find_or_initialize_by({name: value.split.join(" ")})
```
Comment (Magdyx) [inline app/models/book.rb:51]: split?

File: app/models/book.rb [hunk 2]
```diff
@@ -12,11 +12,20 @@ class Book < ApplicationRecord
   before_destroy :cascade
 
   validates :title, presence: true
-  validates :price, presence: true
   validates :description, presence: true
   validates :image_link, presence: true ,:format => URI::regexp(%w(http https))
   validates :isbn, uniqueness:  true
+  validates_numericality_of :isbn
```
Comment (Magdyx) [inline app/models/book.rb:18]: I thought before that the isbn is saved as string.
Anyway, the check length validator uses size method. I suppose it returns the number of bytes in memory not the number of digits. is that correct?

File: app/models/book.rb (hunk 1 above)
Comment (fadyyousry) [reply app/models/book.rb:51]: to avoid extra spaces and ensure the name is unique

File: app/models/book.rb (hunk 2 above)
Comment (fadyyousry) [reply app/models/book.rb:18]: it is a string of digits this validation check that [https://api.rubyonrails.org/classes/ActiveModel/Validations/HelperMethods.html#method-i-validates_numericality_of](url)

File: app/models/book.rb (hunk 1 above)
Comment (Magdyx) [reply app/models/book.rb:51]: if it is only the white spaces use `strip` method, if it is more complex than that give regex a try.

File: app/models/book.rb (hunk 2 above)
Comment (Magdyx) [reply app/models/book.rb:18]: Fair point. My bad, sorry

File: app/models/book.rb (hunk 1 above)
Comment (fadyyousry) [reply app/models/book.rb:51]: is this look better `str.strip.gsub(/\s+/, " ")`?

**Pull request merged**